client/lib/i18n-utils/switch-locale.js

```diff
@@ -33,8 +33,9 @@ function setLocaleInDOM( localeSlug, isRTL ) {
 }
 
 let lastRequestedLocale = null;
-export default function switchLocale( localeSlug ) {
-	const language = getLanguage( localeSlug );
+export default function switchLocale( localeSlug, localeVariant ) {
```

Comment:
Not sure if you had this down for another PR, but are we going consider which json file we want to pull in this function? For example `de_formal.json` etc. 

 My [hacky approach](https://github.com/Automattic/wp-calypso/compare/update/make-switch-locale-support-variant...try/support-locale-variants#diff-8211870589f6cb9c2e411d7115d94c62R51) was to use langSlug of the returned `language`.

Reply:
Nice catch! I've added it in ed4b3d1.